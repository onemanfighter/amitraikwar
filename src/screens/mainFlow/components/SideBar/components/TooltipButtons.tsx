import { SearchIcon, MoonIcon, SunIcon, ThreeDIcon, TwoDIcon } from '@assets';
import {
  Tooltip,
  IconButton,
  useColorMode,
  Box,
  Button,
} from '@chakra-ui/react';
import { modalSelector, useShallow } from '@selectors';
import { ModalIDs, ModalOpenState, appStore } from '@store';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';

const TooltipButtons = () => {
  const { colorMode, toggleColorMode } = useColorMode();
  const [threeDState, setThreeDState] = useState(false);
  const { openModal, closeModal } = appStore(useShallow(modalSelector));
  const { t } = useTranslation();

  const onSearchClick = () => {
    openModal({
      modalOpenState: ModalOpenState.OPEN,
      modalID: ModalIDs.SEARCH,
      onModalClose: closeModal,
    });
  };
  const onDownloadCVClick = () => {
    console.log('Download CV button clicked');
  };
  const on3DToggleClick = () => {
    setThreeDState(!threeDState);
    console.log('3D toggle button clicked');
  };

  return (
    <Box
      margin={'4'}
      padding={'4'}
      gap={'2'}
      flexDirection={'row'}
      justifyContent={'flex-end'}
      alignContent={'center'}
      border={'1px solid gray'}
      borderRadius={'lg'}
    >
      <Button colorScheme="green" marginEnd={8} onClick={onDownloadCVClick}>
        {t('MainHomeScreen.sidebar.downloadCV')}
      </Button>
      <Tooltip
        label={threeDState ? 'Click to make 2D' : 'Click to make 3D'}
        fontSize="xs"
      >
        <IconButton
          aria-label="3D toggle"
          icon={threeDState ? <ThreeDIcon /> : <TwoDIcon />}
          onClick={on3DToggleClick}
          margin={'2'}
        />
      </Tooltip>
      <Tooltip label="Search anything related to me " fontSize="xs">
        <IconButton
          aria-label="Search"
          icon={<SearchIcon />}
          onClick={onSearchClick}
          margin={'2'}
        />
      </Tooltip>
      <Tooltip label={`Click for ${colorMode} mode`} fontSize="xs">
        <IconButton
          aria-label="Toggle dark mode"
          icon={colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
          onClick={toggleColorMode}
          margin={'2'}
        />
      </Tooltip>
    </Box>
  );
};

export default TooltipButtons;
