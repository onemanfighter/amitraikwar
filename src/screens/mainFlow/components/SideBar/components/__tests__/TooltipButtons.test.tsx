import { act, fireEvent, render, screen } from '@testing-library/react';
import TooltipButtons from '../TooltipButtons';

const mockConsole = jest.spyOn(console, 'log');
describe('Tooltip buttons', () => {
  it('should renders correctly', () => {
    const { container } = render(<TooltipButtons />);

    expect(container).toMatchSnapshot();
  });

  it('should print logs on search button click', () => {
    render(<TooltipButtons />);

    fireEvent.click(screen.getByLabelText('Search'));

    expect(mockConsole).toHaveBeenCalledWith('Search button clicked');
  });
});
