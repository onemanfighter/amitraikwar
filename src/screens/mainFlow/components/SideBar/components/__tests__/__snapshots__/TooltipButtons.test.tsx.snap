// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip buttons should renders correctly 1`] = `
<div>
  <div
    class="css-17grvme"
  >
    <button
      class="chakra-button css-1otrqxb"
      type="button"
    >
      Download CV
    </button>
    <button
      aria-label=""
      class="chakra-button css-1xjzpfp"
      type="button"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
    <button
      aria-label="Search"
      class="chakra-button css-1xjzpfp"
      type="button"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
    <button
      aria-label="Toggle dark mode"
      class="chakra-button css-1xjzpfp"
      type="button"
    >
      <svg
        class="w-6 h-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v2.25m6.364.386-1.591 1.591M21 12h-2.25m-.386 6.364-1.591-1.591M12 18.75V21m-4.773-4.227-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
